lib/posix/clock.c

```diff
@@ -25,10 +28,40 @@ static struct k_spinlock rt_clock_base_lock;
  *
  * See IEEE 1003.1
  */
-int z_impl_clock_gettime(clockid_t clock_id, struct timespec *ts)
+int z_impl___posix_clock_get_base(clockid_t clock_id, struct timespec *base)
+{
+	switch (clock_id) {
+	case CLOCK_MONOTONIC:
+		base->tv_sec = 0;
+		base->tv_nsec = 0;
+		break;
+
+	case CLOCK_REALTIME:
+		K_SPINLOCK(&rt_clock_base_lock) {
+			*base = rt_clock_base;
+		}
+		break;
+
+	default:
+		errno = EINVAL;
+		return -1;
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_USERSPACE
+int z_vrfy___posix_clock_get_base(clockid_t clock_id, struct timespec *ts)
+{
+	K_OOPS(K_SYSCALL_MEMORY_WRITE(ts, sizeof(*ts)));
+	return z_impl___posix_clock_get_base(clock_id, ts);
+}
+#include <syscalls/__posix_clock_get_base_mrsh.c>
+#endif
+
+int clock_gettime(clockid_t clock_id, struct timespec *ts)
 {
 	struct timespec base;
-	k_spinlock_key_t key;
 
 	switch (clock_id) {
```

Comment:
Why aren't we replacing this `switch` statement with the `__posix_clock_get_base()`?

Reply:
Because we can avoid incurring the overhead of another `__syscall()` since the monotonic base is always zero.

That's kind of a nice micro-optimization to make queries to the monotonic clock slightly faster.